sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingRelation.scala

```diff
@@ -75,6 +76,52 @@ case class StreamingExecutionRelation(
   )
 }
 
+// We have to pack in the V1 data source as a shim, for the case when a source implements
+// continuous processing (which is always V2) but only has V1 microbatch support. We don't
+// know at read time whether the query is conntinuous or not, so we need to be able to
```

Comment:
can't we know it from the specified trigger?

Reply:
The trigger isn't specified at the point where the dataframe is created.